ted_sws/notice_transformer/services/notice_transformer.py

```diff
@@ -73,26 +87,67 @@ def _write_notice_xml_manifestation_in_package(self, package_path: pathlib.Path,
         with notice_path.open("w", encoding="utf-8") as file:
             file.write(notice.xml_manifestation.object_data)
 
-    def transform_test_data(self, output_path: pathlib.Path):
+    @staticmethod
+    def get_test_notice_container(filename: str) -> str:
+        return NoticeTransformer._get_filename_name(filename)
+
+    @staticmethod
+    def _get_filename_name(filename: str) -> str:
+        """
+        Get filename name without extension
+        :param filename:
+        :return:
+        """
+        return Path(filename).stem
+
+    def _get_out_file_ext(self, filename: str) -> str:
+        """
+        Get file extension, based on output format
+        :param filename:
+        :param ext:
+        :return:
+        """
+        serialization: RMLSerializationFormat = self.rml_mapper.get_serialization_format()
+        exts = {
+            RMLSerializationFormat.TURTLE: '.ttl'
+        }
+        return "{ext}".format(ext=exts.get(serialization, Path(filename).suffix))
+
+    def _write_resource_to_out_file(self, file_resource: FileResource, output_path: Path, idx: int = None):
+        filename = file_resource.file_name
+        notice_container = self.get_test_notice_container(filename)
+        out_filename = notice_container + self._get_out_file_ext(filename)
+        file_resource_parent_path = output_path / Path(notice_container)
+        file_resource_parent_path.mkdir(parents=True, exist_ok=True)
+        file_resource_path = file_resource_parent_path / Path(out_filename)
+        with file_resource_path.open("w+", encoding="utf-8") as f:
+            f.write(file_resource.file_content)
+        if self.loggable:
+            print(("[" + str(idx).rjust(4, "0") + "] " if idx is not None else "") + ":: " + str(
+                datetime.now()
+            ) + " :: " + notice_container, flush=True)
+
+    def transform_test_data(self, output_path: Path):
         """
             This method converts each file in the test data and writes the result to a file in output_path.
         :param output_path:
         :return:
         """
         transformation_test_data = self.mapping_suite.transformation_test_data
-        file_resources = []
-        for data in transformation_test_data.test_data:
-            notice = Notice(ted_id="tmp_notice",xml_manifestation=XMLManifestation(object_data=data.file_content))
+        output_path.mkdir(parents=True, exist_ok=True)
+        test_data = transformation_test_data.test_data
+        if self.loggable:
+            print("")
```

Comment:
we shall have NO print statements anywhere in the code: .... hmmm unless we are in the CLI context, but even there still logging is more appropriate